src/Microsoft.AspNet.Mvc.Core/ApplicationModels/DefaultControllerModelBuilder.cs

```diff
@@ -126,6 +130,12 @@ protected virtual ControllerModel CreateControllerModel([NotNull] TypeInfo typeI
             AddRange(controllerModel.Filters, attributes.OfType<IFilter>());
             AddRange(controllerModel.RouteConstraints, attributes.OfType<IRouteConstraintProvider>());
 
+            var policy = AuthorizationPolicy.Combine(_authorizationOptions, attributes.OfType<AuthorizeAttribute>());
+            if (policy != null)
+            {
+                controllerModel.Filters.Add(new AuthorizeFilter(policy));
```

Comment:
the filter factory is useful since the AuthorizeFilter can directly get the service via a factory. Agreed that this part of the code will still have to do what its doing today.